app/controllers/groups_controller.rb

```diff
@@ -408,6 +391,20 @@ def add_members
     end
   end
 
+  def join
+    ensure_logged_in
+    unless current_user.staff?
+      RateLimiter.new(current_user, "public_group_membership", 3, 1.minute).performed!
+    end
+
+    group = Group.find(params[:id])
+    raise Discourse::InvalidAccess unless group.public_admission
+
+    already_in_group = group.users.exists?(id: current_user.id)
+    return if already_in_group
```

Comment:
This can be a one-liner

```suggestion
    return if already_in_group = group.users.exists?(id: current_user.id)
```

Reply:
I think that a variable with explainable name is better than a comment that can easily become outdated. What do you think?

Reply:
Since the variable is never used, there's not point in declaring and instanciating it.
Up to you if you want to keep the comment. I think it's fine in that case as it makes the code a little bit clearer.

Reply:
Fixed ☑️